### PR TITLE
Moving robot

### PR DESCRIPTION
**Pull request by mohamedsayed18**

### CODE REVIEW

File: tutorials/Moving_robot/moving_robot.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# Moving the robot
+
+In this tutorial we will learn how to move our robot. We will use the robot we built in [Build your own robot](../SDF/car_demo.md) tutorial. You can download the robot from [here](../SDF/car_demo.sdf). You can also find the finished world of this tutorial [here](move_robot.sdf).
```
[the remark this answers is not in the document]
Comment (chapulina) [reply tutorials/Moving_robot/moving_robot.md]: I think we should use the same file, with the same name, across all tutorials

[the remark this answers is not in the document]
Comment (mohamedsayed18) [reply tutorials/Moving_robot/moving_robot.md]: In other tutorial I mentioned that we will rename the file. My motivation for doing so is have the final world file with the same name as the one used in the tutorials

[the remark this answers is not in the document]
Comment (chapulina) [reply tutorials/Moving_robot/moving_robot.md]: humm ok, I need to go over the tutorials again. I'll let you know if it's confusing when I do that, no need to change anything now.

File: tutorials/Moving_robot/moving_robot.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+**Note:** Don't forget to press the play button in the simulation.
+
+The command specifies the topic to publish to after the `-t` option.
+After the `-m` we specify the message type.
+Our robot expects messages of type `Twist` which consists of two components, `linear` and `angular`.
+After the `-p` option we specify the content (value) of the message: linear speed `x: 0.5` and angular speed `z: 0.05`.
+
+**Hint:** You can know what every topic option does using this command: `ign topic -h`
+
+For more information about `Topics` and `Messages` in ignition check the [Transport library tutorials](https://ignitionrobotics.org/api/transport/9.0/tutorials.html)
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:59]:
```suggestion
For more information about `Topics` and `Messages` in Ignition check the [Transport library tutorials](https://ignitionrobotics.org/api/transport/9.0/tutorials.html)
```

File: tutorials/Moving_robot/moving_robot.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
+# Moving the robot
+
+In this tutorial we will learn how to move our robot. We will use the robot we built in the [Build your own robot](../SDF/car_demo.md) tutorial. You can download the robot from [here](../SDF/car_demo.sdf). You can also find the finished world of this tutorial [here](moving_robot.sdf).
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:3]:
```suggestion
In this tutorial we will learn how to move our robot. We will use the
robot we built in the [Build your own robot](../SDF/building_robot.md)
tutorial. You can download the robot from [here](../SDF/building_robot.sdf).
You can also find the finished world of this tutorial [here](moving_robot.sdf).
```

File: tutorials/Moving_robot/moving_robot.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+## What is a plugin
+
+To make our robot move we will use the `diff_drive` plugin. But before doing so let's answer the question "What is a plugin?" A plugin is a chunk of code that is compiled as a shared library and inserted into the simulation. Plugins make us control many aspects of the simulation like world, models etc.
+
+### Diff_drive plugin
+
+`diff_drive` plugin helps us control our robot, specifically a robot that can be differentially driven. Let's setup the plugin on our robot. Open the `car_demo.sdf` and add the following code within the `vehicle_blue` model tags.
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:11]:
```suggestion
`diff_drive` plugin helps us control our robot, specifically a robot that
can be differentially driven. Let's setup the plugin on our robot. Open
the `building_robot.sdf` and add the following code within the `vehicle_blue`
model tags.
```

File: tutorials/Moving_robot/moving_robot.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+Now our model is ready. We just need to send commands (messages) to it.
+These messages will be published (sent) on the `cmd_vel` topic defined above.
+
+A topic is simply a name for grouping a specific set of messages or a particular service.
+Our model will subscribe (listen) to the messages sent on the `cmd_vel` topic.
+
+Launch the car world:
+
+`ign gazebo car_demo.sdf`
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:43]:
```suggestion
`ign gazebo building_robot.sdf`
```

File: tutorials/Moving_robot/moving_robot.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+```xml
+<!-- KeyPublisher plugin-->
+<plugin filename="KeyPublisher" name="Key Publisher"/>
+```
+
+Let's try this plugin as follows:
+
+* In one terminal type
+
+    `ign gazebo car_demo.sdf`
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:79]:
```suggestion
    `ign gazebo building_robot.sdf`
```

File: tutorials/Moving_robot/moving_robot.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    <output type="ignition.msgs.Twist" topic="/cmd_vel">
+        linear: {x: 0.5}, angular: {z: 0.0}
+    </output>
+</plugin>
+```
+
+This code defines the `triggered-publisher` plugin.
+It accepts messages of type `ignition.msgs.Int32` on the `/keyboard/keypress` topic and if the value in the `data` field matches `16777235`(Up arrow key) it outputs a `Twist` message on the `cmd_vel` topic with values `x: 0.5`, `z: 0.0`.
+
+Now launch `car_demo.sdf` and our robot should move forward as we press the Up arrow key &#8593;.
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:115]:
```suggestion
Now launch `building_robot.sdf` and our robot should move forward as we press the Up arrow key &#8593;.
```

File: tutorials/Moving_robot/moving_robot.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+There is a demo explaining how the [Triggered Publisher](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo2/tutorials/triggered_publisher.md) works.
+
+### Moving using arrow keys
+
+To see what values are sent on the `/keyboard/keypress` topic when pressing the arrows we can use the `--echo` or `-e` option
+
+* Run the model in one terminal:
+
+    `ign gazebo car_demo.sdf`
```
Comment (chapulina) [inline tutorials/Moving_robot/moving_robot.md:125]:
```suggestion
    `ign gazebo building_robot.sdf`
```

File: tutorials/Moving_robot/moving_robot.md (hunk 3 above)
Comment (mohamedsayed18) [reply tutorials/Moving_robot/moving_robot.md:3]: Yes, I forget to push it

**Pull request merged**